src/test/java/edu/ucsb/cs156/gauchoride/controllers/RideRequestControllerTests.java

```diff
@@ -0,0 +1,338 @@
+        .day("Monday")
+        .fullName("Bob")
+        .course("ART 1")
+        .startTime("12AM")
+        .stopTime("12AM")
+        .building("Phelps")
+        .room("3525")
+        .pickup("asap")
+        .build();
+    RideRequest u2 = RideRequest.builder()
+        .userid(2L)
+        .day("Monday")
+        .fullName("Victor")
+        .course("ART 1")
+        .startTime("12AM")
+        .stopTime("12AM")
+        .building("Phelps")
+        .room("3525")
+        .pickup("asap")
+        .build();
+
+    ArrayList<RideRequest> expectedRideRequests = new ArrayList<>();
+    expectedRideRequests.addAll(Arrays.asList(u1, u2));
+
+    when(rideRequestRepository.findAll()).thenReturn(expectedRideRequests);
+    String expectedJson = mapper.writeValueAsString(expectedRideRequests);
+    
+    // act
+
+    MvcResult response = mockMvc.perform(get("/api/riderequests/all"))
+        .andExpect(status().isOk()).andReturn();
+
+    // assert
+
+    verify(rideRequestRepository, times(1)).findAll();
+    String responseString = response.getResponse().getContentAsString();
+    assertEquals(expectedJson, responseString);
+
+  }
+
+  @WithMockUser(roles = { "RIDER" })
+  @Test
+  public void rideRequests__rider_succ() throws Exception {
+
+    // arrange
+
+    RideRequest u1 = RideRequest.builder()
+        .userid(1L)
+        .day("Monday")
+        .fullName("Bob")
+        .course("ART 1")
+        .startTime("12AM")
+        .stopTime("12AM")
+        .building("Phelps")
+        .room("3525")
+        .pickup("asap")
+        .build();
+    RideRequest u2 = RideRequest.builder()
+        .userid(2L)
+        .day("Monday")
+        .fullName("Victor")
+        .course("ART 1")
+        .startTime("12AM")
+        .stopTime("12AM")
+        .building("Phelps")
+        .room("3525")
+        .pickup("asap")
+        .build();
+
+    ArrayList<RideRequest> expectedRideRequests = new ArrayList<>();
+    expectedRideRequests.addAll(Arrays.asList(u1));
+
+    when(rideRequestRepository.findAllByUserid(1L)).thenReturn(expectedRideRequests);
+    String expectedJson = mapper.writeValueAsString(expectedRideRequests);
+    
+    // act
+
+    MvcResult response = mockMvc.perform(get("/api/riderequests/all_rider?userid=1"))
+        .andExpect(status().isOk()).andReturn();
+
+    // assert
+
+    verify(rideRequestRepository, times(1)).findAllByUserid(1L);
+    String responseString = response.getResponse().getContentAsString();
+    assertEquals(expectedJson, responseString);
+
+  }
+
+	@WithMockUser(roles = { "RIDER" })
+	@Test
+	public void rider_post() throws Exception {
+			// arrange
+
+			RideRequest u1 = RideRequest.builder()
+        .userid(1L)
+        .day("Monday")
+        .fullName("Bob")
+        .course("ART 1")
+        .startTime("12AM")
+        .stopTime("12AM")
+        .building("Phelps")
+        .room("3525")
+        .pickup("asap")
+        .build();
+			when(rideRequestRepository.save(eq(u1))).thenReturn(u1);
+
```

Comment:
check consistency of indentation throughout the file

Reply:
Will do